mmpose/models/backbones/poseformer.py

```diff
@@ -0,0 +1,312 @@
+        x = x + self.attn(self.norm1(x))
+        x = self.ffn(self.norm2(x), identity=x)
+        return x
+
+
+@BACKBONES.register_module()
+class PoseFormer(BaseBackbone):
+
+    def __init__(self,
+                 num_frame=9,
+                 num_joints=17,
+                 in_chans=2,
+                 embed_dim_ratio=32,
+                 depth=4,
+                 num_heads=8,
+                 mlp_ratio=2.,
+                 qkv_bias=True,
+                 qk_scale=None,
+                 drop_rate=0.,
+                 attn_drop_rate=0.,
+                 drop_path_rate=0.2,
+                 act_cfg=dict(type='GELU'),
+                 norm_cfg=dict(type='LN'),
+                 init_cfg=None):
+        """
+        Args:
+            num_frame (int, tuple): input frame number
+            num_joints (int, tuple): joints number
+            in_chans (int): number of input channels,
+                            2D joints have 2 channels: (x,y)
+            embed_dim_ratio (int): embedding dimension ratio
+            depth (int): depth of transformer
+            num_heads (int): number of attention heads
+            mlp_ratio (int): ratio of mlp hidden dim to embedding dim
+            qkv_bias (bool): enable bias for qkv if True
+            qk_scale (float): override default qk scale
+                              of head_dim ** -0.5 if set
+            drop_rate (float): dropout rate
+            attn_drop_rate (float): attention dropout rate
+            drop_path_rate (float): stochastic depth rate
+            norm_layer: (nn.Module): normalization layer
+        """
+        super().__init__()
+
+        if norm_cfg is None:
+            norm_cfg = dict(type='LN')
+        # Temporal embed_dim is num_joints * spatial embedding dim ratio
+        embed_dim = embed_dim_ratio * num_joints
+        # output dimension is num_joints * 3
+
+        # Spatial patch embedding
+        self.spatial_patch_to_embedding = nn.Linear(in_chans, embed_dim_ratio)
+        self.spatial_pos_embed = nn.Parameter(
+            torch.zeros(1, num_joints, embed_dim_ratio))
+
+        self.temporal_pos_embed = nn.Parameter(
+            torch.zeros(1, num_frame, embed_dim))
+        self.pos_drop = nn.Dropout(p=drop_rate)
+
+        # Stochastic depth decay rule
+        dpr = [x.item() for x in torch.linspace(0, drop_path_rate, depth)]
+
+        self.spatial_blocks = nn.ModuleList([
+            TransformerEncoderLayer(
+                embed_dims=embed_dim_ratio,
+                num_heads=num_heads,
+                feedforward_channels=int(mlp_ratio * embed_dim),
+                qkv_bias=qkv_bias,
+                qk_scale=qk_scale,
+                drop_rate=drop_rate,
+                attn_drop_rate=attn_drop_rate,
+                drop_path_rate=dpr[i],
+                act_cfg=act_cfg,
+                norm_cfg=norm_cfg,
+                init_cfg=init_cfg) for i in range(depth)
+        ])
+
+        self.blocks = nn.ModuleList([
```

Comment:
Does `self.blocks` mean the temporal transformer blocks? How about using `self.temporal _blocks` for clarity?